src/toil/job.py

```diff
@@ -3026,3 +3031,24 @@ def convertPromises(kwargs):
                 foundPromisedRequirement = True
         return foundPromisedRequirement
 
+
+class FileID(str):
+    """
+    A class to wrap the job store file id returned by writeGlobalFile and any attributes we may want
+    to add to it.
+    """
+    def __new__(cls, fileStoreID, *args):
+        return super(FileID, cls).__new__(cls, fileStoreID)
+
+    def __init__(self, fileStoreID, size):
+        super(FileID, self).__init__(fileStoreID)
+        self.size = size
+
+    @classmethod
+    def forPath(cls, fileStoreID, filePath):
+        size = cls._resolveFileSize(filePath)
+        return cls(fileStoreID, size)
+
+    @staticmethod
```

Comment:
I didn't catch this. Separating this out into a method is overkill. Can you fix this in your next PR?